src/main/java/com/arpnetworking/kairosdb/aggregators/HistogramFilterAggregator.java

```diff
@@ -0,0 +1,241 @@
+
+    private static final int PRECISION = 7;
+
+    /**
+     * Whether to keep or discard indeterminate buckets when filtering.
+     */
+    public enum FilterIndeterminate {
+        /**
+         * Keep the individual buckets in the data point group if it is indeterminate if they should be filtered by the
+         * specific query.
+         */
+        KEEP,
+        /**
+         * Discard the individual buckets in the data point group if it is indeterminate if they should be filtered by
+         * the specific query.
+         */
+        DISCARD
+    }
+
+    @FeatureProperty(
+            name = "filter_op",
+            label = "Filter operation",
+            description = "The operation performed for each data point.",
+            type = "enum",
+            options = {"lte", "lt", "gte", "gt", "equal"},
+            default_value = "lt"
+    )
+    private FilterAggregator.FilterOperation _filterop;
+
+    @FeatureProperty(
+            name = "filter_indeterminate_inclusion",
+            label = "Filter indeterminate inclusion",
+            description = "Whether to keep or discard a histogram bin that straddles the threshold when filtering",
+            type = "enum",
+            options = {"keep", "discard"},
+            default_value = "keep"
+    )
+    private FilterIndeterminate _filterinc;
+
+    @FeatureProperty(
+            label = "Threshold",
+            description = "The value the operation is performed on. If the operation is lt, then a null data point "
+                    + "is returned if the data point is less than the threshold."
+    )
+    private double _threshold;
+
+
+    /**
+     * Public Constructor.
+     */
+    @Inject
+    public HistogramFilterAggregator() {
+        _threshold = 0.0;
+        _filterop = FilterAggregator.FilterOperation.LT;
+        _filterinc = FilterIndeterminate.KEEP;
+    }
+
+    public void setFilterOp(final FilterAggregator.FilterOperation filterop) {
+        _filterop = filterop;
+    }
+
+    public void setThreshold(final double threshold) {
+        _threshold = threshold;
+    }
+
+    public void setFilterIndeterminateInclusion(final FilterIndeterminate inclusion) {
+        _filterinc = inclusion;
+    }
+
+    @Override
+    public DataPointGroup aggregate(final DataPointGroup dataPointGroup) {
+        return new HistogramFilterDataPointAggregator(dataPointGroup);
+    }
+
+    @Override
+    public boolean canAggregate(final String groupType) {
+        return HistogramDataPointFactory.GROUP_TYPE.equals(groupType);
+    }
+
+    @Override
+    public String getAggregatedGroupType(final String groupType) {
+        return HistogramDataPointFactory.GROUP_TYPE;
+    }
+
+    static double truncate(final double val) {
+        final long mask = 0xffffe00000000000L;
+        return Double.longBitsToDouble(Double.doubleToRawLongBits(val) & mask);
+    }
+
+    private static double binBound(final double val) {
```

Comment:
nit: could you comment or rename this to make it clear that it's an _inclusive_ bound?